numba/tests/test_typedlist.py

```diff
@@ -689,6 +826,319 @@ def foo():
         self.assertEqual(0, got[0])
 
 
+class TestDisableReflectedListBase(MemoryLeakMixin, TestCase):
+
+    def _njit_both(self, func):
+        return (njit(func, _disable_reflected_list=p) for p in (True, False))
+
+
+class TestListBuiltinConstructors(TestDisableReflectedListBase):
+
+    def test_simple_refine_list_builtin(self):
+        def foo():
+            l = list()
+            l.append(1)
+            return l
+        foo_true, foo_false = self._njit_both(foo)
+        foo_true_received, foo_false_received = foo_true(), foo_false()
+        foo_true, foo_false = self._njit_both(foo)
+        foo_true_received, foo_false_received = foo_true(), foo_false()
+        foo_true_expected, foo_false_expected = List(), list([1])
+        foo_true_expected.append(1)
+        self.assertEqual(foo_true_expected, foo_true_received)
+        self.assertEqual(foo_false_expected, foo_false_received)
+        self.assertEqual(List, type(foo_true_received))
+        self.assertEqual(list, type(foo_false_received))
```

Comment:
Put this into a `self.check(foo_true_expected, foo_true_received)`? It occurs repeatedly.

Reply:
yes